### PR TITLE
Refactor render error message#12745

### PR DESCRIPTION
**Pull request by Clark-W-Griswold**
## Screenshots
before:
image

after:
image

## Acceptance criteria
- [x] Improved automated test(s): thoroughness (of use/edge-cases) is more important than line coverage
- [x] PR created
- [x] PR reviewed (tag @cvalarida as a PR reviewer)
- [ ] PR merged 
- [ ] Smoke-tested on staging ([verification steps](url))
- [ ] Review staging analytics (verify previous weeks to analyze trends)
- [ ] Pre-flight check: If issue, triage and assert roll forward/roll back. 
- [ ] Prod deployment   
- [ ] Smoke-tested on prod ([verification steps](url)) 
- [ ] Review production analytics (verify daily over next few days)

### CODE REVIEW

File: src/platform/forms-system/src/js/review/submit-states/GenericError.jsx [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
+import { Column, Row } from 'platform/forms/components/common/grid';
 
 export default function GenericError(props) {
   const { appType, formConfig, renderErrorMessage, onSubmit, testId } = props;
   let submitButton;
   let submitMessage;
 
   if (renderErrorMessage) {
-    submitMessage = renderErrorMessage();
+    submitMessage = <SavableErrorMessage formConfig={formConfig} />; // renderErrorMessage();
```
Comment (Clark-W-Griswold) [inline src/platform/forms-system/src/js/review/submit-states/GenericError.jsx:17]: Will remove the comment in the next commit.

File: src/platform/forms/components/review/SavableErrorMessage.jsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
+// libs
```
Comment (Clark-W-Griswold) [inline src/platform/forms/components/review/SavableErrorMessage.jsx:1]: We still need to break up this component to be more modular. This is more of the first steps to get there.

File: src/platform/forms/components/review/SavableErrorMessage.jsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        <p>
+          If it still doesn’t work, please <CallHRC />
+        </p>
+      );
+    }
+
+    return (
+      <ErrorMessage
+        active
+        title={`We’re sorry. We can't submit your ${appType} right now.`}
```
Comment (cvalarida) [inline src/platform/forms/components/review/SavableErrorMessage.jsx:75]:
```suggestion
        title={`We’re sorry. We can’t submit your ${appType} right now.`}
```

File: src/platform/forms/components/review/SavableErrorMessage.jsx [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+// platform - forms - selectors
+import {
+  formSelector,
+  showLoginModalSelector,
+  userSelector,
+} from 'platform/forms/selectors/review';
+import { defaultProps } from 'recompose';
+
+function SavableErrorMessage(props) {
```
Comment (cvalarida) [inline src/platform/forms/components/review/SavableErrorMessage.jsx:32]: What do you think about the name `FormSaveErrorMessage`? The error message isn't savable itself; it's reporting an error on the form save.

File: src/platform/forms/components/review/SavableErrorMessage.jsx [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+  const saveLink = (
+    <SaveFormLink
+      locationPathname={location?.pathname}
+      form={form}
+      formConfig={route?.formConfig}
+      user={user}
+      showLoginModal={showLoginModal}
+      saveAndRedirectToReturnUrl={defaultProps.saveAndRedirectToReturnUrl}
+      toggleLoginModal={defaultProps.toggleLoginModal}
```
Comment (cvalarida) [inline src/platform/forms/components/review/SavableErrorMessage.jsx:48]: Is this how `defaultProps` is used? I've never used it myself, but it doesn't look the same as the [API docs](https://github.com/acdlite/recompose/blob/master/docs/API.md#defaultprops). Looks like `defaultProps` is a function used in conjunction with [`compose`](https://github.com/acdlite/recompose/blob/master/docs/API.md#compose) to create higher-order components. :confused:

Comment (cvalarida) [reply src/platform/forms/components/review/SavableErrorMessage.jsx:48]: According to the [`recompose` project page](https://github.com/acdlite/recompose), it looks like `recompose` is depreciated in favor of hooks.

>Hooks solves all the problems I attempted to address with Recompose three years ago, and more on top of that. I will be discontinuing active maintenance of this package (excluding perhaps bugfixes or patches for compatibility with future React releases), and recommending that people use Hooks instead.

How would we use hooks here instead? Do we need to? Should these `defaultProps` be `props`?

Is this tested? (I realize this gets into the weird space of component integration testing.)

Comment (Clark-W-Griswold) [reply src/platform/forms/components/review/SavableErrorMessage.jsx]: I think this was an auto import. I have removed this as it was not intended.

File: src/platform/forms/components/review/SavableErrorMessage.jsx (hunk 4 above)
Comment (Clark-W-Griswold) [reply src/platform/forms/components/review/SavableErrorMessage.jsx]: Sounds good, name updated.

File: src/platform/forms/components/review/FormSaveErrorMessage.jsx [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+        <p>
+          If it still doesn’t work, please <CallHRC />
+        </p>
+      );
+    }
+
+    return (
+      <ErrorMessage
+        active
+        title={`We’re sorry. We can't submit your ${appType} right now.`}
```
Comment (cvalarida) [inline src/platform/forms/components/review/FormSaveErrorMessage.jsx:74]:
```suggestion
        title={`We’re sorry. We can’t submit your ${appType} right now.`}
```

File: src/platform/forms-system/test/js/review/submit-states/GenericError.unit.spec.jsx [hunk 7]
```diff
@@ -199,23 +199,23 @@ describe('Schemaform review: <GenericError />', () => {
...
+          renderErrorMessage
           testId="12345"
         />
       </Provider>,
     );
 
-    expect(tree.getByText('Error message')).to.not.be.null;
+    expect(
+      tree.getByText(
+        "We’re sorry. We can't submit your application right now.",
```
Comment (cvalarida) [inline src/platform/forms-system/test/js/review/submit-states/GenericError.unit.spec.jsx:216]:
```suggestion
        "We’re sorry. We can’t submit your application right now.",
```

File: src/platform/forms/tests/components/review/SavableErrorMessage.unit.spec.jsx [hunk 8]
```diff
@@ -0,0 +1,122 @@
+// libs
```
Comment (cvalarida) [inline src/platform/forms/tests/components/review/SavableErrorMessage.unit.spec.jsx:1]: Probably makes sense to rename this unit test file to `FormSaveErrorMessage.unit.spec.jsx` now.

**Review by cvalarida — APPROVED** [review]
I left a couple of clean-up comments, but it otherwise looks great. :shipit:
